### ISSUE TITLE
[Enhancement] Separate routing and modal

### ISSUE THREAD

**Issue by kodeaben** [kirbydesign/designsystem#2902]
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the enhancement
As the implementation of modal is right now, the implementer will always have to import a `RouterModule` because the `ModalController` requires the `ActivateRoute` to be provided.

The issue is that not all implementation options of `Modal` uses a route.

The implementer are therefore in some cases forced to import a module which is not needed for the usecase. 
An example for this is the first example in the [cookbook -> modal](https://cookbook.kirby.design/#/home/showcase/modal) (the one called Inside the parent (caller) component). It looks like this:

```Typescript
constructor(private modalController: ModalController) {}

showModal() {
  const config: ModalConfig = {
    flavor: 'modal',
    component: YourEmbeddedModalComponent,
    componentProps: {
      prop1: 'value1',
      prop2: 'value2'
    }
  };
  this.modalController.showModal(config);
}
```

If you implement this in a project which doesn't use routing (and therefore havn't imported `RouterModule`), you'll get the following error: 
![image](https://user-images.githubusercontent.com/114976818/221581258-61c0e633-550e-48fd-ac5d-7d443f8b2d59.png)

NB! This issue doesn't exist in [modal experimental](https://cookbook.kirby.design/#/home/showcase/modal-experimental), but it is also present in `Page`. An Issue about Pages use of the `RouterModule`/`ActivatedRoute` can be found here: #2903 

## Describe the solution you'd like
The best solution would to have routing as a separate part of the modal, so the example above could work.
## Have you considered any alternatives?
Yes, but not any good ones.

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Refinement

- [-] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-refinement); do not proceed until this is done.
- [ ] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".

**Comment by kodeaben** [comment]
This issue is closed, because it is only present in the current version of `Modal` , but not in the `modal/experimental` which will replace the current one in the nearest future.
As a part of this investigation, it was found that `Page` also has a dependency on `ActivatedRoute`. An issue about this can be found here: #2903
